tests/system/conftest.py

```diff
@@ -34,3 +35,81 @@ def identity():
         channel=grpc.insecure_channel('localhost:7469'),
     )
     return IdentityClient(transport=transport)
+
+
+class _GenericClientInterceptor(grpc.UnaryUnaryClientInterceptor,
+                                grpc.UnaryStreamClientInterceptor,
+                                grpc.StreamUnaryClientInterceptor,
+                                grpc.StreamStreamClientInterceptor):
+
+    def __init__(self, interceptor_function):
+        self._fn = interceptor_function
+
+    def intercept_unary_unary(self, continuation, client_call_details, request):
+        new_details, new_request_iterator, postprocess = self._fn(
+            client_call_details, iter((request,)), False, False)
+        response = continuation(new_details, next(new_request_iterator))
+        return postprocess(response) if postprocess else response
```

Comment:
Style nit: it may make the code a little clearer to lift this evaluation into its own line, i.e.
```python
response = postprocess(response) if postprocess else response
return response
```
Judgement call on your part, feel free to ignore.